openpype/pipeline/farm/pyblish_functions.py

```diff
@@ -620,15 +620,26 @@ def _create_instances_for_aov(instance, skeleton, aov_filter, additional_data,
         aov_patterns = aov_filter
 
         preview = match_aov_pattern(app, aov_patterns, render_file_name)
-        # toggle preview on if multipart is on
-        if instance.data.get("multipartExr"):
-            log.debug("Adding preview tag because its multipartExr")
-            preview = True
 
         new_instance = deepcopy(skeleton)
         new_instance["subset"] = subset_name
         new_instance["subsetGroup"] = group_name
 
+        # toggle preview on if multipart is on
+        # Because we cant query the multipartExr data member of each AOV we'll
+        # need to have hardcoded rule of excluding any renders with
+        # "crytomatte" in the file name from being a multipart EXR. This issue
+        # happens with Redshift that forces Cryptomatte renders to be separate
+        # files even when the rest of the AOVs are merged into a single EXR.
+        # There might be an edge case where the main instance has cryptomatte
+        # in the name even though its a multipart EXR.
+        if (instance.data.get("multipartExr") and
+           "cryptomatte" not in render_file_name.lower()):
+            log.debug("Adding preview tag because its multipartExr")
+            preview = True
+        else:
+            new_instance["multipartExr"] = False
+
```

Comment:
This shouldn't be here - you're trying to solve something for Maya + Redshift but this happens for ALL renderers on potentially all platforms going through this function. This has the dangerous side effect of potentially influencing way more than you intended it to.

If you want to exclude cryptomatte AOVs **by name** for your reviews/previews then you should use the `aov_filters` regex in settings to adjust what names should included for previews or not. Not sure if it works if it's still marked multipartExr but we use `project_settings/deadline/publish/ProcessSubmittedJobOnFarm/aov_filter` 

We have quite some madness there in the regex currently to exclude any `crypto` layer and/or `Z` layer (because it didn't have RGB channels it failed, it only had Z channel). For example with negative lookaheads, etc.

```
^(?:(?!([._][Cc]rypto|\.)|_Z[._]).)*\.\d+\.[A-Za-z]+$
```

---

If it's really specific (only when multipartEXR is enabled, etc.) then I think this should be in a Maya collector for Redshift setting that particular resulting instance to not be multi part exr, or whatever?

Anyway, this should not be hardcoded here.

Reply:
> We have quite some madness there in the regex currently to exclude any crypto layer and/or Z layer (because it didn't have RGB channels it failed, it only had Z channel). For example with negative lookaheads, etc.

This does not work for me.

> If it's really specific (only when multipartEXR is enabled, etc.) then I think this should be in a Maya collector for Redshift setting that particular resulting instance to not be multi part exr, or whatever?

The issue is that the instances are not generated until the deadline publish submission; https://github.com/ynput/OpenPype/blob/develop/openpype/modules/deadline/plugins/publish/submit_publish_job.py#L393

As I said in the description, ideally we should have separate instances at the collection stage, but that is too big of a scope for this PR.

How about we make it specific to `redshift` renderer as I've done now?

Reply:
Fine with me. Thanks for confirming.